tensorflow/python/data/ops/dataset_ops.py

```diff
@@ -1007,8 +1007,22 @@ def i(...):
       return ParallelMapDataset(self, map_func, num_parallel_calls)
 
   def flat_map(self, map_func):
-    """Maps `map_func` across this dataset and flattens the result.
+    """Maps `map_func` across this dataset and flattens the result. 
+    
+    `tf.data.Dataset.interleave()` is a generalization of `flat_map`, since 
+    `flat_map` produces a similar outputs as `tf.data.Dataset.interleave(cycle_length=1)`
+    
+    Use `flat_map` if you want to make sure, that the order of your dataset stays the same.
+    For example, to implement unbatch:
```

Comment:
Thanks for adding an example. Please move it above the mention of "interleave".

I'd suggest replacing the words "to implement unbatch" with "to flatten a dataset of batches into a dataset of their elements".